pp.c

```diff
@@ -4868,6 +4868,94 @@ PP(pp_fc)
     RETURN;
 }
 
+PP(pp_trim)
+{
+    dSP;
+    dTARGET;
+    SV *source = TOPs;
+    STRLEN len;
+    const U8 *start;
+    SV *dest;
+
+    SvGETMAGIC(source);
+
+    if(SvOK(source))
+        start = (const U8*)SvPV_nomg_const(source, len);
```

Comment:
Should this die() if a reference is passed? It’d be a nice convenience, anyway.

Reply:
No, making this the only string operator that did that would be strange.

Also, checking for a ref here would be more than checking for SvROK(), you'd also need to check if there's overloading magic.

The real solution to that type of type check would be a (default on) feature I think.

Reply:
@tonycoz Are you talking about something that would throw on such operations as:

- `[] =~ m//`
- substr( {}, 0, 2 )
- trim( \123 )

… and the like?

Reply:
Yes, so:

```
package Foo {
  use overload '""' => sub { "hello" };
}
package Bar {} # no string overloading
my $foo = bless {}, "Foo";
my $bar = bless {}, "Bar";
my $x = substr({}, 0, 1); # doesn't throw
my $y = "abc";
print \$y; # doesn't throw
print $foo; # doesn't throw
print $bar; # doesn't throw
no feature 'ref_as_string'; # or whatever we call it
print \$y; # throws
$x = substr({}, 0, 1); # throws
print $foo; # doesn't throw
print $bar; # throws
```

There's corner cases I haven't covered (globs?)